openedx/adg/lms/webinars/views.py

```diff
@@ -50,6 +56,71 @@ def get_context_data(self, **kwargs):
             'registration': registration,
             'is_registered': is_registered,
             'send_post_call_to_register': send_post_call_to_register,
-            'featured_speakers': self.object.featured_speakers
+            'featured_speakers': self.object.featured_speakers,
+            'google_url': self.get_google_calendar_url(),
+            'yahoo_url': self.get_yahoo_calendar_url(),
         })
         return context
+
+    def get_google_calendar_url(self):
+        """
+        Generate url for the webinar to add the reminder on the google calendar
+
+        Returns:
+            str: google link
+        """
+        base_url = 'https://www.google.com/calendar/event?'
+        end_time = self.object.start_time + self.object.duration
+        details = {
+            'action': 'TEMPLATE',
+            'text': self.object.title,
+            'details': self.object.description,
+            'dates': self.object.start_time.strftime('%Y%m%dT%H%M%S') + '/' + end_time.strftime('%Y%m%dT%H%M%S'),
+            'ctz': pytz.utc
+        }
+        return base_url + urllib.parse.urlencode(details)
+
+    def get_yahoo_calendar_url(self):
+        """
+        Generate url for the webinar to add the reminder on the yahoo calendar
+
+        Returns:
+            str: yahoo link
+        """
+        base_url = 'https://calendar.yahoo.com/?v=60&'
+        end_time = self.object.start_time + self.object.duration
+        details = {
+            'title': self.object.title,
+            'desc': self.object.description,
+            'st': self.object.start_time.strftime('%Y%m%dT%H%M%SZ'),
+            'et': end_time.strftime('%Y%m%dT%H%M%SZ'),
+            'ctz': pytz.utc
+        }
+        return base_url + urllib.parse.urlencode(details)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Attaches .ics for webinar to download with response.
+
+        Returns:
+            HttpResponse: Response with .ics file.
+        """
+        if not request.GET.get('download_ical'):
+            return super().get(request, *args, **kwargs)
+
+        webinar = self.get_object()
+
+        cal = Calendar()
+        event = Event()
+        event.add('summary', webinar.title)
+        event.add('description', webinar.description)
+        event.add('dtstart', webinar.start_time)
+        event.add('duration', webinar.duration)
+        event.add('location', webinar.meeting_link)
+        event.add('uid', uuid.uuid4())
+        cal.add_component(event)
+
+        response = HttpResponse(cal.to_ical(), content_type='text/calendar; charset=utf8')
+        response['Content-Disposition'] = 'attachment; filename=calendar.ics'
```

Comment:
Wouldn't it be better to include title of the webinar in the downloaded filename?